tests/unit_tests/test_unit.py

```
@@ -2461,6 +2461,65 @@ def test_spectrum_preprocessing(tmp_path, mgf_small):
+
+def test_log_annotate_report_missing_predictions(monkeypatch):
+    mock_logger = unittest.mock.MagicMock()
+    monkeypatch.setattr("casanovo.utils.logger", mock_logger)
+    monkeypatch.setattr(utils, "log_run_report", lambda **_: None)
+    utils.log_annotate_report([], n_missing_predictions=4)
+    calls = [
+        c
+        for c in mock_logger.info.call_args_list
+        if "did not receive a prediction" in c[0][0]
+    ]
+    assert calls and calls[0][0][1] == 4
```

Ruff (0.16.2)

[warning] Assertion should be broken down into multiple parts

Break down assertion into multiple parts

(PT018)